modules/javafx.graphics/src/test/java/test/javafx/stage/Popup_owner_Test.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2011, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.stage;
+
+import java.util.ArrayList;
+import java.util.stream.Stream;
+
+import javafx.scene.Node;
+import javafx.stage.Stage;
+import javafx.stage.Popup;
+import javafx.beans.InvalidationListener;
+import javafx.beans.Observable;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import test.com.sun.javafx.test.objects.TestGroup;
+import test.com.sun.javafx.test.objects.TestNode;
+import test.com.sun.javafx.test.objects.TestScene;
+import test.com.sun.javafx.test.objects.TestStage;
+
+
+/**
+ * This test specifically checks for ownerNode and ownerWindow properties.
+ * Because those properties are set via different show() methods instead of
+ * regular setter method, we can't use PropertiesTestBase for this check.
+ * Instead we rely on checking this manually.
+ */
+public final class Popup_owner_Test {
```

Comment:
minor: maybe `Popup_Owner_Test`?

Reply:
We do seem to have this funny-looking `PascalCase_camelCase_Test` convention in several places.

Reply:
Yeah I mostly followed the convention of `Popup_parentWindow_Test`. I assumed the original followed `ClassToTest_propertyName_Test` convention and since this one tests `ownerWindow` and `ownerNode` properties at the same time, the name got reduced to `owner`.